src/scripts/main.js

```diff
@@ -1,3 +1,381 @@
+labelAge.appendChild(inputAge);
+labelSalary.appendChild(inputSalary);
+
+form.append(
+  labelName,
+  labelPosition,
+  labelCountry,
+  labelAge,
+  labelSalary,
+  button,
+);
+
+document.body.appendChild(form);
+
+tableRows.forEach((tr) => {
+  tr.children[0].setAttribute('data-row', 'name');
+  tr.children[1].setAttribute('data-row', 'position');
+  tr.children[2].setAttribute('data-row', 'office');
+  tr.children[3].setAttribute('data-row', 'age');
+  tr.children[4].setAttribute('data-row', 'salary');
+});
+
+const showNotification = (message, type) => {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.style.display = 'flex';
+  div.style.justifyContent = 'center';
+  div.style.alignItems = 'center';
+  div.classList.add('notification', type);
+  div.textContent = message;
+  document.body.appendChild(div);
+
+  setTimeout(() => {
+    div.remove();
+  }, 3000);
+};
+
+const validateInput = (obj) => {
+  if (obj.name.length < 4 || !obj.name.trim()) {
+    showNotification('Name must be at least 4 letters', 'error');
+
+    return false;
+  }
+
+  if (obj.position.length === 0 || !obj.position.trim()) {
+    showNotification('Position is required', 'error');
+
+    return false;
+  }
+
+  if (obj.age < 18 || obj.age > 90) {
+    showNotification('Age must be between 18 and 90', 'error');
+
+    return false;
+  }
+
+  if (obj.salary.length === 0 || obj.salary < 0) {
+    showNotification('Salary is required', 'error');
+
+    return false;
+  }
+
+  return true;
+};
+
+const addComa = (value) => {
+  const improvedValue = ('' + value).split('').reverse();
+  let result = '';
+
+  for (let i = 0; i < improvedValue.length; i++) {
+    if (improvedValue.length >= 3) {
+      if (i % 3 === 0) {
+        result += ',';
+      }
+      result += improvedValue[i];
+    } else {
+      return value;
+    }
+  }
+
+  result = result.split('').reverse().join('');
+
+  if (result.endsWith(',')) {
+    result = result.slice(0, -1);
+  }
+
+  return result;
+};
+
+button.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const formDataObj = Object.fromEntries(formData);
+
+  if (validateInput(formDataObj)) {
+    const tr = document.createElement('tr');
+
+    for (const key in formDataObj) {
+      const td = document.createElement('td');
+
+      td.setAttribute('data-qa', key);
+      td.setAttribute('data-row', key);
+
+      const value = formDataObj[key];
+
+      if (key === 'salary') {
+        td.textContent = `$${addComa(value)}`;
+      } else {
+        td.textContent = value;
+      }
+      tr.appendChild(td);
+    }
+    tbody.appendChild(tr);
+    showNotification('You added the new employee', 'success');
+    editRow(tr);
+
+    for (const input of form) {
+      input.value = '';
+    }
+  }
+  updateTableRows();
+});
+
+const updateTableRows = () => {
+  tableRows = [...document.querySelectorAll('tbody tr')];
```

Comment:
The syntax for updating `tableRows` is incorrect. The square brackets should not be there. It should be `document.querySelectorAll('tbody tr')` without the brackets.